File: papi/lista1/exercicio10.py
while True:
    try:
        nota1 = float(input("Por favor digite sua nota 1: "))
        break
    except ValueError:
        print("O valor deve ser um número válido, com ponto como separador decimal")

while True:
    try:
        nota2 = float(input("Por favor digite sua nota 2: "))
        break
    except ValueError:
        print("O valor deve ser um número válido, com ponto como separador decimal")

while True:
    try:
        nota3 = float(input("Por favor digite sua nota 3: "))
        break
    except ValueError:
        print("O valor deve ser um número válido, com ponto como separador decimal")

print("A média das notas é", (nota1 + nota2 + nota3) / 3)

File: papi/lista1/exercicio10.kt
package lista1

fun main() {
    println("Iremos calcular a média de 3 notas")
    var notas: ArrayList<Double> = ArrayList()
    
    var cont = 1
    while (cont < 4) {
        while (true) {
            try {
                print("Digite a nota $cont: ")
                val nota = readLine()!!.toDouble()
                if (nota < 0) {
                    throw NumberFormatException()
                }
                notas.add(nota)
                break
            } catch (e: NumberFormatException) {
                println("Por favor digite um número maior que zero válido, usando o ponto como separador decimal")
            }
        }
        cont = cont + 1
    }

    print("A média das notas é: ")
    println(notas.average())
}